src/scielo/bin/xml/app_modules/generics/dbm/dbm_isis.py

```diff
@@ -76,50 +72,40 @@ def tag_data(self, tag, data):
         if not isinstance(data, list):
             data = [data]
         for item in data:
-            occs.append(self.tag_occ(tag, item))
+            occs.append(self._tag_occ(tag, item))
         return occs
 
-    def tag_occ(self, tag, data):
-        s = u''
-        if isinstance(data, tuple):
-            encoding.debugging('tag_occ()', tag)
-            encoding.debugging('tag_occ()', data)
-        elif isinstance(data, dict):
-            s = self.tag_content(tag, self.format_subfields(data))
-        else:
-            s = self.tag_content(tag, format_value(data))
-        return s
-
-    def format_subfield(self, subf, subf_value):
-        res = u''
-        if subf in 'abcdefghijklmnopqrstuvwxyz123456789':
-            res = '^' + subf + change_circ(format_value(subf_value))
-        elif subf != '_':
-            encoding.debugging('format_subfield()', ('ERR0', subf, subf_value))
-        return res
-
-    def format_subfields(self, subf_and_value_list):
-        first = u''
-        value = u''
-        values = []
+    def _tag_occ(self, tag, data):
+        """
+        Cria cada ocorrência de um dado campo
+        """
+        if not data:
+            return ""
         try:
-            first = subf_and_value_list.get('_', u'') or u''
-            values = sorted([self.format_subfield(k, v) for k, v in subf_and_value_list.items() if v is not None and len(v) > 0])
-            value = u''.join(values)
-        except Exception as e:
-            encoding.report_exception('format_subfields()', e, subf_and_value_list)
-            encoding.report_exception('format_subfields()', e, (first, values))
+            data = {"_": data + ""}
+        except TypeError:
+            if not isinstance(data, dict):
+                raise TypeError("IDFile.tag_occ expects dict or str")
+        return self._tag_content(tag, self._format_subfields(data))
```

Comment:
Este `return` deve acontecer mesmo se o `TypeError` acontecer e `data` não for um `dict`? Talvez precise de um else nesse `try`.

Reply:
return deve acontecer mesmo se o TypeError acontecer e data **for** um